libs/store_helper/ebpf_store_helper.c

```diff
@@ -0,0 +1,352 @@
+
+        // Save helper count.
+        status = write_registry_value_dword(
+            program_key, EBPF_PROGRAM_DATA_HELPER_COUNT, program_info[i].count_of_program_type_specific_helpers);
+        if (!IS_SUCCESS(status)) {
+            close_registry_key(program_key);
+            goto Exit;
+        }
+
+        if (program_info[i].count_of_program_type_specific_helpers != 0) {
+            // Create (or open) helper registry path.
+            status = create_registry_key(
+                program_key, EBPF_PROGRAM_DATA_HELPERS_REGISTRY_PATH, REG_CREATE_FLAGS, &helper_info_key);
+            if (!IS_SUCCESS(status)) {
+                close_registry_key(program_key);
+                goto Exit;
+            }
+
+            // Iterate over all the helper prototypes and save in registry.
+            for (uint32_t count = 0; count < program_info[i].count_of_program_type_specific_helpers; count++) {
+                status = ebpf_store_update_helper_prototype(
+                    helper_info_key, &(program_info[i].program_type_specific_helper_prototype[count]));
+                if (!IS_SUCCESS(status)) {
+                    close_registry_key(program_key);
+                    close_registry_key(helper_info_key);
+                    goto Exit;
+                }
+            }
+
+            close_registry_key(helper_info_key);
+        }
+        close_registry_key(program_key);
+    }
+
+Exit:
+    if (program_info_key) {
+        close_registry_key(program_info_key);
+    }
+    if (provider_key) {
+        close_registry_key(provider_key);
+    }
+
+    return status;
+}
+
+/**
+ * @brief Update global helper information in eBPF store.
+ *
+ * @param[in] helper_info Pointer to an array of helper function prototypes.
+ * @param[in] helper_info_count Count of helper function prototypes.
+ *
+ * @returns Status of the operation.
+ */
+__return_type
+ebpf_store_update_global_helper_information(
+    _In_reads_(helper_info_count) ebpf_helper_function_prototype_t* helper_info, uint32_t helper_info_count)
+{
+    __return_type status = _SUCCESS;
+    ebpf_registry_key_t provider_key = NULL;
+    ebpf_registry_key_t helper_info_key = NULL;
+
+    if (helper_info_count == 0) {
+        return status;
+    }
+
+    // Open (or create) provider registry path.
+    status = ebpf_store_open_or_create_provider_registry_key(&provider_key);
+    if (!IS_SUCCESS(status)) {
+        goto Exit;
+    }
+
+    // Open (or create) global helpers registry path.
+    status = create_registry_key(provider_key, EBPF_GLOBAL_HELPERS_REGISTRY_PATH, REG_CREATE_FLAGS, &helper_info_key);
+    if (!IS_SUCCESS(status)) {
+        goto Exit;
+    }
+
+    for (uint32_t i = 0; i < helper_info_count; i++) {
+
+        status = ebpf_store_update_helper_prototype(helper_info_key, &helper_info[i]);
+        if (!IS_SUCCESS(status)) {
+            goto Exit;
+        }
+    }
+
+Exit:
+    if (helper_info_key) {
+        close_registry_key(helper_info_key);
+    }
```

Comment:
For functions we implement it's better to allow passing in null rather than doing such `if` checks.  It makes the code smaller and increases code coverage numbers to pass in null.

Reply:
Agree, will optimize the existing implementation.